doc/syntax/comments.rdoc

```diff
@@ -35,3 +35,193 @@ syntax error:
     Will not work
     =end
   end
+
+== Magic Comments
+
+While comments are typically ignored by Ruby, special "magic comments" contain
+directives that affect how the code is interpreted.
+
+Top-level magic comments must start on the first line, or on the second line if
+the first line looks like <tt>#! shebang line</tt>.
+
+NOTE: Magic comments affect only the file in which they appear;
+other files are unaffected.
+
+  # frozen_string_literal: true
+
+  var = 'hello'
+  var.frozen? # => true
+
+=== Alternative syntax
+
+Magic comments may consist of a single directive (as in the example above).
+Alternatively, multiple directives may appear on the same line if separated by ";"
+and wrapped between "-*-" (See Emacs' {file variables}[https://www.gnu.org/software/emacs/manual/html_node/emacs/Specifying-File-Variables.html])
+
+  # -*- emacs-compatible; encoding: big5; frozen_string_literal: true -*-
```

Comment:
`emacs-compatible` part is invalid as a file variable.
And `encoding` is not a pre-defined variable then asked if apply that variable.

```suggestion
  # emacs-compatible; -*- coding: big5; mode: ruby -*-
```